### ISSUE TITLE
Not able to enter data into a content type stored on Postgres 11

### ISSUE THREAD

**Issue by tshelver** [strapi/strapi#3009]
Node: v11.12.0
Npm: V6.7.0
Strapi: V3.0.0-alpha.25.2
Ubuntu 18.04.
DB: Postgresql 11.2
What is the current behavior?

Created content type.  1 relation category, 1 relation product, joined to the category.
Created against Postgresql 11,
Tried to add a new category, failed
Error as below:
<<<<<<<<<<<<
[2019-03-20T19:04:47.661Z] debug GET /content-manager/explorer/product?_limit=20&_start=0&source=content-manager (9 ms)
[2019-03-20T19:04:47.677Z] debug GET /content-type-builder/autoReload (2 ms)
[2019-03-20T19:04:47.679Z] debug GET /settings-manager/autoReload (2 ms)
[2019-03-20T19:04:47.768Z] debug GET /content-manager/models (5 ms)
[2019-03-20T19:05:40.526Z] error insert into "category" ("created_at", "desc", "name", "updated_at") values ($1, $2, $3, $4) returning "id" - relation "category" does not exist
[2019-03-20T19:05:40.526Z] debug POST /content-manager/explorer/category/?source=content-manager (21 ms)

Changed the connection back to default Mongo, and it works.....  Assuming that it is really Mongo under the hood.

What is the expected behavior?
That it words?
Suggested solutions
Wait for beta, or better yet, production...

**Comment by tshelver** [comment]
BTW, the only thing stored in postgres was a table 'categories_products__products_categories......... No data in it.

**Comment by tshelver** [comment]
Started a new project.  Configured to use Postgres from the beginning, rather than adding afterwards. Now it works...

So there is a workaroudn, should probably be documented.  Seems like the ability to add a different database after project creation is not working yet

**Comment by tshelver** [comment]
Further on this.  
Starting strapi with Postgrsql today, Warning that indexes were not created as per below See at bottom of my comments.

I investigated the situation.  

- ID  columns were are automatically generated as primary keys.  Correct.
There is NO need for an index to be created here by strapi as Postgres will automatically generated. So no warming necessary

- There was an index created against the foreign key lookup.  However... This is a gin index against the NAME of the object, not the id column which should be a btree index.  Very inefficient.  Not sure what happens when the name is changed.

- The issue of adding Postgres to a project generated the standard way doesn't seem to work, as documented above.  Even deleting the content types, or starting new content types and allocating them to the Postgres connection, results in join tables ONLY being generated in Postgres. The other tables remained in Mongo. 

I have no desire to implement, learn or administer Mongo, one of the key selling points for Strapi to me was Postgres support.

Startup code follows (note that it is started in debug mode despite the warning that further info is only available in debug mode.

<<<<<<<<<<<<<<<<<
[2019-03-21T07:41:57.509Z] warn The SQL database indexes haven't been generated successfully. Please enable the debug mode for more details.
[2019-03-21T07:41:57.511Z] warn The SQL database indexes haven't been generated successfully. Please enable the debug mode for more details.
[2019-03-21T07:41:58.018Z] info Time: Thu Mar 21 2019 09:41:58 GMT+0200 (South Africa Standard Time)
[2019-03-21T07:41:58.018Z] info Launched in: 10818 ms
[2019-03-21T07:41:58.019Z] info Environment: development
[2019-03-21T07:41:58.019Z] info Process PID: 15178
[2019-03-21T07:41:58.019Z] info Version: 3.0.0-alpha.25.2 (node v11.12.0)
[2019-03-21T07:41:58.019Z] info To shut down your server, press <CTRL> + C at any time

[2019-03-21T07:41:58.020Z] info ☄️  Admin panel: http://localhost:1337/admin
[2019-03-21T07:41:58.020Z] info ⚡ Server: http://localhost:1337

[2019-03-21T07:43:24.068Z] debug GET index.html (83 ms)
[2019-03-21T07:43:24.249Z] debug GET /favicon.ico (7 ms)
[2019-03-21T07:43:34.143Z] debug GET index.html (28 ms)

**Comment by lauriejim** [comment]
Hello @tshelver !
Can you give me exact step to reproduce your issue.
Especially for your content type definition, it's not clear.

**Comment by tshelver** [comment]
2 different issues.  
Issue 1.
If you start a new project, and take the recommended setting, the project is initiated with Mongo as a default.
If you then add in Postgres as an extra connection, and create a new content type ( used category and product, with a many-to-many relationship from Category to Product, only the resolving table (category_product__product_category is created in Postgres.    It appears that the relations / tables are still in Mongo.

**Comment by tshelver** [comment]
Issue 2.
This is the last issue I commented on.  I created a new project, and selected 'custom' option, so that I could select Postgres from the getgo.  This does not create a Mongo connection.
But if you add the same 2 tables / relations (this time with a 1-many relationship to make things simpler), the tables are created correctly in Postgres, along with the necessary  indexes.

However, to me it seems a bit strange to create  the foreign key index on the 2 'name' attributes as a default, and not on the strapi-generated primary key columns.  And especially to use a gin index, which is not as efficient as a btree.

However, I think I am giving up on Postgres for now.  I added data to the two content types, tried to add extra fields using the content type manager, and the system refused to restart (which it does after content type changes).

I have another outstanding issue which is a much bigger roadblock.  If that is resolved, maybe I will come back to experiment more with PG.
